test/animedb.auth.api.test.js

```
@@ -0,0 +1,232 @@
+const chai = require('chai');
+const assert = chai.assert;
+const chaiHttp = require('chai-http');
+const mongoose = require('mongoose');
+const app = require('../lib/app');
+const connection = require('../lib/mongooseSetup');
+chai.use(chaiHttp);
+mongoose.Promise = Promise;
+
+describe('Checking access to anime DB with user auth enabled', done => {
+    before( done => {
+        const dropAnimes = () => {
+            connection.db.dropCollection('animechars', (err, result) => {
+                if (err) done(err);
+                else connection.db.dropCollection('animeshows', (err2, result2) => {
+                    if (err2) done(err2);
```

These error blocks aren't really useful. `dropCollection` will throw an error if the collection doesn't already exist, and you're probably not really interested in that. You just want it dropped if it does exist. So ignore the error, or maybe log it out. It will only appear the first time you run the tests on a new DB. And at some point you'll probably want to start dropping the test DB after running the tests, which would make these lines crash your test every time.